lib/addExcursionPointers.m

```diff
@@ -0,0 +1,132 @@
+%==========================================================================
+%This function takes in an NIDM-Results json and creates a hashmap labelling
+%objects with a pointer to their relevant excursion set. The resultant
+%hashmap takes in the id of an object from the main NIDM-Results graph and
+%outputs a number for it's corresponding ExcursionMap.
+%
+%graph - the graph from an NIDM-Results json.
+%
+%Authors: Thomas Maullin, Camille Maumet.
+%==========================================================================
+
+function labels = addExcursionPointers(graph)
+    
+    %Create an empty list for the keys and values.
+    labelValues = {};
+    labelKeys = {};
+    
```

Comment:
The reason I've done it this way around instead of adding keys and values individually is I *think* that it's meant to be more time efficient as it prevents the hashmap resizing itself when too many key-value pairs are added... I'm not 100% though - I can change it if you'd like?